tezos/context/src/serialize/mod.rs

```diff
@@ -278,53 +278,61 @@ pub fn deserialize_hash_id(data: &[u8]) -> Result<(Option<HashId>, usize), Deser
     let byte_hash_id = data.get(0).copied().ok_or(UnexpectedEOF)?;
 
     if byte_hash_id & 1 << 7 != 0 {
-        // The HashId is in 3 bytes
-        let hash_id = data.get(0..3).ok_or(UnexpectedEOF)?;
+        // The HashId is in 4 bytes
```

Comment:
@sebastiencs so, serialized hashids will be 4 bytes until they don't fit there anymore? how is this tested?

Reply:
@tizoc Yes, `HashId` will be 4 bytes as long as they fit in 31 bits. Starting with 32 bits, they will be serialized on 6 bytes.
The tests for this are there: https://github.com/tezedge/tezedge/pull/854/files#diff-eba41f64fb01eb676751ac27bd6a3a994a283b4e5ec57fb7af8977ad4d31a37cR1678

Before this PR, there is a similar behavior where HashId are serialized in 3 or 4 bytes.